### PR TITLE
Data Sources: Remove Admin/Data sources page in favour of Connections/Data sources

### PR DESCRIPTION
**Pull request by mikkancso**
**What is this feature?**

This PR removes the **Admin/Data sources** page and introduces a redirection to the **Connections/Data sources** page.
The same is true for
- New datasource page (datasources/new)
- Edit datasource page (datasources/edit)
- Datasource dashboards page (datasources/edit/:uid/dashboards)

I didn't set up a redirect for the datasources details page, because we would have to redirect from `plugins/:id`, but it's not only data sources that are served at that URL. So we would have to detect if that ID is of a data source, and then redirect, which feels a little bit twisted.

Demo:
[Screencast from 2023-07-17 11-13-11.webm](https://github.com/grafana/grafana/assets/13637610/ef9b5513-4375-4384-991a-b298e9567004)

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/cloud-onboarding/issues/4008

**Special notes for your reviewer:**
I informed the docs team about this change. I'm not sure about the status of this on their side, though.

There is something wrong with the navigation around the datasource dashboards page. In fact, it is already broken on `main`: https://github.com/grafana/grafana/issues/71843
I tried to fix it, but couldn't.

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/admin/components/PluginListItem.tsx [hunk 1]
```diff
@@ -21,8 +22,13 @@ export function PluginListItem({ plugin, pathName, displayMode = PluginListDispl
   const styles = useStyles2(getStyles);
   const isList = displayMode === PluginListDisplayMode.List;
 
+  const href =
+    plugin.type === PluginType.datasource
+      ? locationUtil.assureBaseUrl(`${CONNECTIONS_ROUTES.DataSources}/${plugin.id}`)
+      : `${pathName}/${plugin.id}`;
+
```
Comment (jackw) [inline public/app/features/plugins/admin/components/PluginListItem.tsx:29]: The jump from plugins catalog to connections from a user perspective feels rather jarring/confusing as the side navigation and breadcrumbs change and forces the user to the browser back button as the only way to return to where they were. Is this change required to manage a datasource instance? If not I would vote that we don't add this right now. @sympatheticmoose WDYT?

Comment (sympatheticmoose) [reply public/app/features/plugins/admin/components/PluginListItem.tsx:29]: Agreed 👍 I think we still remove the duplicate pages issue without requiring this redirect at this stage, and this could add more confusion than value.

Comment (mikkancso) [reply public/app/features/plugins/admin/components/PluginListItem.tsx:29]: :ok_hand: I'll revert this change

File: pkg/services/navtree/navtreeimpl/admin.go [hunk 2]
```diff
@@ -18,16 +17,6 @@ func (s *ServiceImpl) getAdminNode(c *contextmodel.ReqContext) (*navtree.NavLink
 	orgsAccessEvaluator := ac.EvalPermission(ac.ActionOrgsRead)
 	authConfigUIAvailable := s.license.FeatureEnabled("saml")
 
-	if hasAccess(datasources.ConfigurationPageAccess) {
-		configNodes = append(configNodes, &navtree.NavLink{
-			Text:     "Data sources",
-			Icon:     "database",
-			SubTitle: "Add and configure data sources",
-			Id:       "datasources",
-			Url:      s.cfg.AppSubURL + "/datasources",
```
Comment (jackw) [inline pkg/services/navtree/navtreeimpl/admin.go:27]: Given this is a minor release I'm not sure if this should be deleted or the url changed to point to connections page so anyone that is used to finding the datasources list via admin navigation section can still get to it. @sympatheticmoose WDYT?

**Review by jackw — APPROVED** [review]
LGTM! 🚀 

Having reviewed it again I've left another nit regarding UX.

File: pkg/services/navtree/navtreeimpl/admin.go (hunk 2 above)
Comment (mikkancso) [reply pkg/services/navtree/navtreeimpl/admin.go:27]: Well, on the other hand if I click on something under the Admin section, I expect to land in the Admin section. Redirection at that point feels also very strange to me - not the best UX IMO.

Comment (sympatheticmoose) [reply pkg/services/navtree/navtreeimpl/admin.go:27]: Both fair points. But:
- I don't think we want to wait till Grafana 11 to remove it
- We've at least advised that the data sources page was effectively deprecated with the warning banner
- If we changed the URL only for now, whilst its a more gentle removal, it feels similar to the initial redirect from clicking a datasource card.

Comment (mikkancso) [reply pkg/services/navtree/navtreeimpl/admin.go:27]: Thank you all for your inputs! I'll merge this then :)
